dungeon/src/contrib/systems/EventScheduler.java

```diff
@@ -0,0 +1,101 @@
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class is responsible for scheduling and executing timed actions. It maintains a list of
+ * scheduled actions, each of which is represented by a ScheduledAction object that encapsulates a
+ * IAction and an execution time. The class provides methods to schedule new actions, clear all
+ * scheduled actions, and update the state of scheduled actions.
+ *
+ * @see IAction
+ */
+public class EventScheduler extends System {
+
+  private static final List<ScheduledAction> scheduledActions = new ArrayList<>();
+
+  /**
+   * Schedules a new action to be executed after a specified delay.
+   *
+   * <p>This method creates a new ScheduledAction with the provided action and execution time, and
+   * adds it to the list of scheduled actions. The execution time is calculated as the current time
+   * plus the provided delay.
+   *
+   * @param action The action to be executed.
+   * @param delayMillis The delay in milliseconds after which the action should be executed.
+   * @return The scheduled action that was created and added to the list.
+   */
+  public static ScheduledAction scheduleAction(IAction action, long delayMillis) {
+    long executeAt = TimeUtils.millis() + delayMillis;
+    ScheduledAction scheduledAction = new ScheduledAction(action, executeAt);
+    scheduledActions.add(scheduledAction);
+    return scheduledAction;
+  }
+
+  /**
+   * Clears all scheduled actions.
+   *
+   * <p>This method removes all actions from the list of scheduled actions. After this method is
+   * called, the list of scheduled actions will be empty.
+   */
+  public static void clear() {
+    scheduledActions.clear();
+  }
+
+  /**
+   * Cancels specific scheduled actions from the list.
+   *
+   * <p>This method removes the specified scheduled action from the list of scheduled actions. If
+   * the action is not found in the list, it will not be removed.
+   *
+   * @param scheduledAction The scheduled actions to be removed.
+   * @return The removed scheduled actions.
+   */
+  public static ScheduledAction[] cancelAction(ScheduledAction... scheduledAction) {
+    for (ScheduledAction action : scheduledAction) {
+      scheduledActions.remove(action);
+    }
+    return scheduledAction;
+  }
+
+  /**
+   * Executes the scheduled actions.
+   *
+   * <p>This method is called to process all the scheduled actions. It first gets the current time,
+   * then it starts iterating over the scheduled actions. For each scheduled action, it checks if
+   * the current time is greater than or equal to the execution time of the action. If it is, it
+   * applies the effect of the action and removes the action from the list of scheduled actions.
+   * After all actions have been processed, it ends the iteration over the scheduled actions.
+   */
+  @Override
+  public void execute() {
+    long currentTime = TimeUtils.millis();
+    if (scheduledActions.isEmpty()) {
+      return;
+    }
```

Comment:
das ist irgendwie überflüssig, oder? wenn die liste leer ist, dann wird die `for`-loop anschließend ja sowieso nicht ausgeführt?